### PR TITLE
Enhancement/3236 update wait for

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #3236

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/e2e/specs/modules/optimize/activation.test.js [hunk 1]
```diff
@@ -61,21 +61,26 @@ async function proceedToOptimizeSetup() {
...
-		),
-	] );
-	await expect( page ).toMatchElement(
-		'.googlesitekit-publisher-win__title',
-		{ text: /Congrats on completing the setup for Optimize!/i }
-	);
-}
-
 describe( 'Optimize Activation', () => {
+	async function finishOptimizeSetup() {
```
Comment (eugene-manuilov) [inline tests/e2e/specs/modules/optimize/activation.test.js:65]: I can't explain why this function needs to be moved here, but it fixes timeout issues. 🤷

**Review by tofumatt — COMMENTED** [review]
This looks great, and I don't think I have any requested changes but a few questions about the PR and some changes before we merge. Looks great though!

And I looked through all the VRT changes; they seem good 👍🏻

File: assets/js/components/Button.stories.js [hunk 2]
```diff
@@ -64,16 +64,14 @@ IconButton.storyName = 'Icon Button with Tooltip';
...
 };
 
 export const VRTStory = () => (
 	<div>
 		<p>
 			<DefaultButton { ...DefaultButton.args } />
 		</p>
-		<p>
-			<HoverButton { ...HoverButton.args } />
-		</p>
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/components/Button.stories.js:76]: Yep, this hover button state doesn't make sense for VR tests anyway :smile:

File: package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
 		"build:dev": "webpack --mode=development --devtool cheap-module-eval-source-map --output-pathinfo",
 		"prebuild:production": "npm run remove-dist",
 		"build:production": "webpack --mode=production",
-		"build:test": "npm run build:production -- --include-tests",
+		"build:test": "npm run build:dev -- --include-tests",
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply package.json]: I had issues with e2e tests on my local and updated this command to build assets in the development mode. Now, this is not needed anymore. Reverted.

File: jest-puppeteer.config.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ module.exports = {
 			'--no-sandbox',
 			'--disable-setuid-sandbox',
 			'--disable-dev-shm-usage',
+			'--disable-web-security',
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply jest-puppeteer.config.js]: Added

**Review by tofumatt — APPROVED** [review]
Awesome, thanks! 👍🏻

**Pull request merged**